### DISCUSSION TITLE
[KEP] Allow using `let` to define local variable in "legacy `for` loop" like Javascript

### DISCUSSION THREAD

**Discussion by anzhi0708** [kuroko-lang/kuroko#36] [Ideas]
*I can't find suitable title... so I'll go with 'KEP' (Kuroko Enhancement Proposal) lol*

Using `var` defines global and  `let` defines local variable in nodeJS (`for` loop)
![IMG_20221020_105504](https://user-images.githubusercontent.com/57821133/196837906-76107a91-5cf2-4bd0-a8b8-29eabd9956fb.jpg)
They act differently.

Well, just personal thoughts though ;-)

**Comment by anzhi0708** [comment]
Before:
```
for _ in range(5):
    let i = _       # catch the value
    ...             # do stuff
```
After:
```
for let i = 0; i < 5; i++:  # init local variable i
    ...  # do stuff
```
...or even...
```
for let i in range(5):
    ...
```

**Comment by klange** [comment]
I don't want to use issues for enhancement proposals, so I've turned on Discussions - one of the default categories is "Ideas", let's put this sort of thing there.

(Also I really have to resist the urge to come up with an alternative to "proposal" that starts with "k" so we can have "KEKs"... good thing I can't find a sensible one...)
